controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -118,8 +121,10 @@ func (r *StorageClassRequestReconciler) Reconcile(ctx context.Context, request r
 
 	result, reconcileError = r.reconcilePhases()
 
-	// Apply status changes to the StorageClassRequest
-	statusError := r.Client.Status().Update(r.ctx, r.StorageClassRequest)
+	controllerutil.AddFinalizer(r.StorageClassRequest, v1alpha1.StorageClassRequestFinalizer)
+
+	// Apply status changes and finalizer to the StorageClassRequest
+	statusError := r.Client.Update(r.ctx, r.StorageClassRequest)
```

Comment:
You should not update that status using `r.Client.Update`. Please revert the change.